core/java/android/view/ViewRootImpl.java

```diff
@@ -4204,6 +4205,19 @@ protected void onDeliverToNext(QueuedInputEvent q) {
         }
 
         protected boolean shouldDropInputEvent(QueuedInputEvent q) {
+            boolean isDesktopInteractive =
+                    SystemProperties.getBoolean(PROPERTY_MARUOS_DESKTOP_INTERACTIVE, false);
+            // We will drop all input events except touch screen when Linux is enable.
+            // In this occasion, we will use (bluetooth or usb) keyboard and mouse
+            // in Linux, and use touch screen/pad/navigation in Android for controlling
+            // the state of Linux manually when using Android cast to show Linux.
+            boolean isTouchEvent = q.mEvent != null
```

Comment:
From my testing it looks like the Android software "back" button is being ignored when the desktop input is enabled. Maybe it is falling into this check?

Reply:
Yes, the software `back` button is a physical event, so I change the logic to drop external input events and keep all built-in input events which will fix this problem and simplify the code.